core/src/main/java/org/apache/iceberg/rest/ErrorHandlers.java

```diff
@@ -0,0 +1,137 @@
+
+import java.io.IOException;
+import java.util.function.Consumer;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.core5.http.HttpStatus;
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.iceberg.exceptions.AlreadyExistsException;
+import org.apache.iceberg.exceptions.BadRequestException;
+import org.apache.iceberg.exceptions.ForbiddenException;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.exceptions.NotAuthorizedException;
+import org.apache.iceberg.exceptions.RESTException;
+import org.apache.iceberg.exceptions.ServiceFailureException;
+import org.apache.iceberg.rest.responses.ErrorResponse;
+
+/**
+ * A set of consumers to handle errors for requests for table entities or for namespace entities,
+ * to throw the correct exception.
+ */
+public class ErrorHandlers {
+
+  private ErrorHandlers() {
+  }
+
+  /**
+   * Table level error handlers.
+   * Should be chained wih the {@linkplain ErrorHandlers#defaultErrorHandler}, which takes care of common cases.
+   */
+  public static Consumer<ErrorResponse> tableErrorHandler() {
+    return error -> {
+
+      switch (error.code()) {
+        // table resource routes can encounter namespace not found or table not found.
+        // TODO - Need to define response types (can probably use the HTTP values and specialize from there).
+        case HttpStatus.SC_NOT_FOUND:
+          // TODO - error.type() could be null for HEAD responses. We probably want to pass
+          //        in some request info using a BiConsumer so we can have the table name etc
+          //        from the request.
+          if (NoSuchNamespaceException.class.getSimpleName().equals(error.type())) {
+            throw new NoSuchNamespaceException("Resource not found: %s", error);
+          } else {
+            throw new NoSuchTableException("Resource not found: %s", error);
+          }
+        case HttpStatus.SC_CONFLICT:
+          throw new AlreadyExistsException("The table already exists: %s", error);
+        default:
+          throw new RESTException("Unknown error: %s", error);
+      }
+    };
+  }
+
+  /**
+   * Request error handlers specifically for CRUD ops on namespaces.
+   * Should be chained wih the {@linkplain ErrorHandlers#defaultErrorHandler}, which takes care of common cases.
+   */
+  public static Consumer<ErrorResponse> namespaceErrorHandler() {
+    return error -> {
+      switch (error.code()) {
+        case HttpStatus.SC_NOT_FOUND:
+          throw new NoSuchNamespaceException("Namespace not found: %s", error);
+        case HttpStatus.SC_CONFLICT:
+          throw new AlreadyExistsException("Namespace already exists: %s", error);
+        // Invalid UpdateNamespacePropertiesRequest - keys to remove overlapped with keys of properties to add
```

Comment:
What does this comment mean? Isn't this what `SC_UNPROCESSABLE_ENTITY` means? ([spec](https://github.com/apache/iceberg/blob/master/docs/rest/rest-catalog-open-api.yaml#L303))

Reply:
The comment is mostly to clarify why the exception message is so specific.